### ISSUE TITLE
Load multiple csv files into one datatable and select columns

### ISSUE THREAD

**Issue by Peter-Pasta** [h2oai/datatable#2613]
Lets say I have a list with filenames (all csv files) and want to load them into one single datatable and also select only some columns. The R code I use is the following:

`df <- files_selected %>% map_df(~fread(., select = c("c1", "c2", "c3")))`

How would I achieve the same in Python?

Thank you very much

**Comment by samukweku** [comment]
@Peter-Pasta you can use the [iread](https://datatable.readthedocs.io/en/latest/api/dt/iread.html) function to read in a list of files; the following code should get the job done : 

```python
dataframes = iread(files_selected, columns={'c1','c2','c3'}) # passing a set of the columns picks out only the specified columns
#combine into one 
dt.rbind(tuple(dataframes))
```

I will try and include scenarios such as this into the documentation. Let me know if you still have any issues.

Cheers

**Comment by oleksiyskononenko** [comment]
@Peter-Pasta I would just add to the @samukweku's answer that it is better to immediately assign the output of the `dt.rbind()` to a new frame. This is because the output of `dt.iread()` is an iterator, that can only be iterated once.

So would recommend to do something like that

```
from datatable import dt
DT_iterator = dt.iread(FILES_LIST, columns=COLUMNS_LIST) 
DT = dt.rbind(list(DT_iterator))
```

**Comment by Peter-Pasta** [comment]
@oleksiyskononenko and @samukweku, I tried the suggested code but I can't get it working. This is the error I get:

I use the datatable versions 0.10.1
<img width="462" alt="Bildschirmfoto 2020-09-14 um 10 29 55" src="https://user-images.githubusercontent.com/67552385/93062779-ab135980-f675-11ea-81e1-2e1484d7f95f.png">

<img width="843" alt="Bildschirmfoto 2020-09-14 um 10 23 26" src="https://user-images.githubusercontent.com/67552385/93062192-e5c8c200-f674-11ea-9ead-0bdeade4a5ca.png">
<img width="534" alt="Bildschirmfoto 2020-09-14 um 10 23 33" src="https://user-images.githubusercontent.com/67552385/93062205-e8c3b280-f674-11ea-8ed7-b6051cb60609.png">

Thank you very much for your help so far :)

**Comment by samukweku** [comment]
Try installing the development version, [iread](https://datatable.readthedocs.io/en/latest/releases/v0.11.0.html#fread) should be available in version 0.11. If, however, you cannot install the development version, you can iterate with ``fread``:

```python
# passing a set to the columns argument picks out only the specified columns
# from each file
dataframes = [fread(file, columns={'c1','c2','c3'}) for file in files_selected]
#combine into one dataframe
combo = dt.rbind(dataframes)
```

**Comment by oleksiyskononenko** [comment]
@Peter-Pasta Oh, the problem here is that `iread()` was only added recently and it was not a part of the `0.10.1` release. We're going to release a new version `0.11.0` pretty soon and `iread()` will be a part of it. So I guess you have several options:
1) Build a dev version of datatable
2) Wait a little bit, then upgrade datatable with pip
3) Do what @samukweku suggests to make it work right away

**Comment by Peter-Pasta** [comment]
@oleksiyskononenko and @samukweku, thank you very much. I will use @samukweku code and just wait for version 0.11.0. Any additions to the documentation would be extremely useful. For example based on what I read in the docs new columns are computed the following way:

`DT[:, {"x": f.x, "y": f.y, "x+y": f.x + f.y, "x-y": f.x - f.y}]`

But in my case this just prints the new column to the console instead of adding it to the datatable. Is this also something that is dependent on the version of datatable or am I missing something?

**Comment by samukweku** [comment]
You need to reassign the outcome to `DT`.

**Comment by Peter-Pasta** [comment]
@samukweku, currently I use the following code:

`df = dt.cbind(df, df[:, {"new_column": f.old_column}])`

That seems not very elegant to me and secondly in R it automatically assigns the outcome to the df:

`df[, new_column := old_column]` 

I think I wouldn't be the only one who would find an explanation of this behavior in the docs very helpful.

Thank you both very much for your advice.

**Comment by samukweku** [comment]
@st-pasha , in `datatable`, there is the `update` method, which is similar to R's walrus operator : 

```python
df[:, update(new_column =  f.old_column)]
```
Your initial solution is elegant, efficient/fast is probably what you have in mind. Note also, that ``update`` operates in-place; you do not need to reassign the outcome.

**Comment by st-pasha** [comment]
If the idea is to create a new column that is a copy of an existing column, then imho the simplest way is 
```
df["new_column"] = f.old_column
```

If, on the other hand, you want to change the name of an existing column, then you can assign to `.names` property:
```
df.names = {"old_column": "new_column"}
```

**Comment by Peter-Pasta** [comment]
@st-pasha the new column is dependent on an existing column either based on a calculation or some sort of string slicing. Regarding the string slicing I actually wanted to ask when this will be supported. I saw this was already requested (#1667) and is listed in the Roadmap (#2281) and Refactor Expr architecture (#2562). 

I really appreciate your help.

**Comment by st-pasha** [comment]
The RHS of assignment can be any expression:
```
df["new_column"] = (f.old_column + f.very_old_column) / f.ancient_column - f.primordial_column
```

As for string slicing feature - it is still not implemented, but you can :+1: the original issue #1667 to indicate your interest.

**Comment by samukweku** [comment]
@Peter-Pasta , is there any part of this issue that is unresolved? If there is, could you kindly indicate that? If there isn't, do you mind closing the issue? Thanks  and please keep sending in your suggestions and feedback.
